temporian/implementation/numpy/data/dtype_normalization.py

```diff
@@ -114,6 +114,21 @@ def normalize_features(
     `normalize_features` should match `_DTYPE_MAPPING`.
     """
 
+    def _to_bytes_array(
+        feat_array: Union[Tuple, List, np.ndarray]
+    ) -> np.ndarray:
+        """Try to convert string/object array to np.bytes or fail"""
+        try:
+            if isinstance(feat_array, np.ndarray):
+                return feat_array.astype(np.bytes_)
+            return np.array(feat_array, dtype=np.bytes_)
+        except UnicodeEncodeError as e:
+            raise ValueError(
+                f"Feature {name} contains non-supported unicode characters."
```

Comment:
In your example, the issue was with non-ascii characters (!= non-unicode).
Maybe we can add support for unicode characters easily?